Client/Experiments/Experiments.swift

```diff
@@ -217,6 +217,37 @@ enum Experiments {
 
         log.info("Nimbus is initializing!")
     }
+    
+    public static func initializeFirstRun(_ options: InitializationOptions) {
+        let nimbus = Experiments.shared
+
+        nimbus.initialize()
+
+        switch options {
+        case .preload(let url): nimbus.setExperimentsLocally(url)
+        case .testing(let payload): nimbus.setExperimentsLocally(payload)
+        default: break /* noop */
+        }
+
+        // on first run, we want to get the experiments from remote settings
+        // first
+        let start = DispatchTime.now()
+        NotificationCenter.default.addObserver(forName: .nimbusExperimentsApplied, object: nil, queue: .main) { _ in
+            let end = DispatchTime.now()
+            let timeElapsed = end.uptimeNanoseconds - start.uptimeNanoseconds
+            let timeInterval = Double(timeElapsed) / 1_000_000_000 // Technically could overflow for long running tests
+            log.info("Time elapsed before available: \(timeInterval) seconds!")
+        }
+        nimbus.fetchExperiments()
+        NotificationCenter.default.addObserver(forName: .nimbusExperimentsFetched, object: nil, queue: .main) { _ in
```

Comment:
Eeeh, the order here should probably be flipped (i.e the observer added before the fetch) but I'll do that once this is something we actually wanna merge